CONTRIBUTING.md

```diff
@@ -0,0 +1,118 @@
+- Connaisseur is successfully installed in a test cluster
+- a non-signed image is deployed to the cluster and denied
+- a signed image is deployed to the cluster and passed
+- Connaisseur is successfully uninstalled
+
+We hope to automate these tests in the future and also provide a test script for local testing with Minikube.
+
+### Signed Commits and Pull Requests
+All changes to the `master` branch must be signed which is enforced via [branch protection](https://docs.github.com/en/free-pro-team@latest/github/administering-a-repository/about-required-commit-signing). This can be achieved by only fast-forwarding signed commits or signing of merge commits by a contributor. Consequently, while we appreciate signed commits in PRs, we do not require it.
+
+A general introduction into signing commits can for example be found in the [With Blue Ink blog](https://withblue.ink/2020/05/17/how-and-why-to-sign-git-commits.html). For details on setting everything up for GitHub, please follow the steps in the [Documentation](https://docs.github.com/en/github/authenticating-to-github/managing-commit-signature-verification).
+
+Once you have generated your local GPG key, added it to your GitHub account and informed Git about it, you are setup to create signed commits. We recommend to configure Git to sign commits by default via:
+```bash
+git config commit.gpgsign true
+```
+This avoids forgetting to use the `-S` flag when committing changes. In case it happens anyways, you can always rebase to sign earlier commits:
+```bash
+git rebase -i master
+```
+You can then mark all commits that need to be signed as `edit` and sign them without any other changes via:
+```bash
+git commit -S --amend --no-edit
+```
+Finally, you force push to overwrite the unsigned commits via `git push -f`.
+
+### Semantic and Conventional Commits
+For Connaisseur, we want to use semantic and conventional commits to ensure good readability of code changes. A good introduction to the topic can be found in [this blog post](https://nitayneeman.com/posts/understanding-semantic-commit-messages-using-git-and-angular/).
+
+Commit messages should consist of header, body and footer. Such a commit message takes the following form:
+```
+git commit -m "<header>" -m "<body>" -m "<footer>"
+```
+The three parts should consist of the following:
+- _header_: Comprises of a commit type (common types are described below) and a concise description of the actual change, e.g. `fix: extend registry validation regex to custom ports`.
+- _body_ (optional): Contains information on the motivation behind the change and considerations for the resolution, `The current regex used for validation of the image name does not allow using non-default ports for the image repository name. The regex is extended to optionally provide a port number.`.
+- _footer_ (optional): Used to reference PRs, issues or contributors and mark consequences such as breaking changes, e.g. `Fix #<issue-number>`
```

Comment:
Do `Fix #<issue-number>` and `Fixes #<issue-number>` behave the same? We only tested `Fixes ...`, didn't we?

Reply:
Yes, according to https://guides.github.com/features/issues/#notifications :
```
By prefacing your commits with “Fixes”, “Fixed”, “Fix”, “Closes”, “Closed”, or “Close” when the commit is merged into main, it will also automatically close the issue.
```